src/main/java/com/aws/greengrass/util/platforms/SystemResourceController.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.util.platforms;
+
+import com.aws.greengrass.lifecyclemanager.GenericExternalService;
+
+import java.util.Map;
+
+public interface SystemResourceController {
+    /**
+     * Remove the resource controller. This method should be called when an existing generic external service is
+     * removed.
+     *
+     * @param component a generic external service
+     */
+    void removeResourceController(GenericExternalService component);
```

Comment:
should the component definitely need to be a generic external service?

Reply:
GreengrassService, just in case we want to add anything else which is a sibling to GES instead of a child.